_posts/2024-08-28-Data-Prepper-2.9.0-is-ready-for-download.md

```diff
@@ -0,0 +1,120 @@
+---
+
+## Introduction
+
+You can download Data Prepper 2.9.0 today.
+This feature release includes a number of core improvements and improvements to many popular processors.
+
+
+## Improvements to expressions
+
+Data Prepper continues to improve support for expressions to allow you more control for routing and when conditions.
+In this release, Data Prepper adds support for set operations.
+These operations allow you to write conditions that check that a value is in a set of possible values.
+This can be especially useful for routing where you need to route data depending on the system that data comes from.
+
+Additionally, Data Prepper now has a new `startsWith` function to determine if a string value starts with another string.
+
+## Default route
+
+Data Prepper has had sink routing since the 2.0 release.
+With this capability, pipeline authors can use Data Prepper expressions to route events to different sinks to meet their requirements.
+One challenge that pipeline authors had was how to handle events that did not match any existing routes.
+A common solution that teams found to work-around this was to have a route that was the inverse of other routes.
+But, this required that they copy the other conditions and invert it.
+This was difficult to handle and even harder to maintain.
+
+Now, Data Prepper supports a special route named `_default`.
+By applying this route to a sink, pipeline authors can ensure that events that do not match any other routes will go a default sink of their choosing.
+
+For example, consider a simple situation where you want to route front-end and back-end events to different sinks.
+You can define two sinks for these and define your routes.
+But, what if you receive events that do not match?
+The sample pipeline below shows you how you could handle events that do not match either the front-end or back-end routes.
+
+```
+routes:
+  - frontend: '/service == "front-end"`
+  - backend: '/service == "back-end"`
+sink:
+ - opensearch:
+      routes:
+         - front-end
+ - opensearch:
+      routes:
+         - back-end
+  - opensearch:
+       routes:
+          - _default
+```
+
+## Performance
+
+The Data Prepper maintainers have been working toward improving the performance of Data Prepper.
+This release includes a number of internal improvements that speed up processing for many processors.
+You don't have to do anything other than update your version to get these improvements.
+
+Data Prepper 2.9 also offers some new features that you can use to help reduce out-of-memory errors or circuit breaker trips.
+Many pipelines involve taking some source data in the form of a string and then extracting that data into a structure.
+Some examples of these are `grok` and `parse_json`.
+When you use these processors, you more than double the size of each event you are processing.
+As the events flowing through the system are the largest part of memory usage, this will greatly increase your memory requirements.
```

Comment:
```suggestion
Because the events flowing through the system are the largest part of memory usage, this will greatly increase your memory requirements.
```